### PR TITLE
first cut of Windows secret store

### PR DESCRIPTION
**Pull request by zanderz**
@patrickxb The tests pass but I realize that if we ship this as-is, people will be prompted to log in again, leaving the unlocked file-based creds laying around, right? should we do something about that?

### CODE REVIEW

**Comment by maxtaco** [comment]
Can't you store the creds from the `.ss` file into the windows password manager, and then delete the `.ss` file?

**Comment by patrickxb** [comment]
I would do something like max suggested.  Create a secret store file via `NewSecretStoreFile(g.Env.GetDataDir())`, retrieve the secret from it, store it in the windows secret store, then clear the secret from SecretStoreFile.

I'm working on an improved secret store file in my current branch, so use RetrieveSecret and ClearSecret and don't mess with the files directly otherwise it won't work.

**Comment by patrickxb** [comment]
This looks good to me.  I think some real-world testing before this gets released would be a good idea to make sure the upgrade from file to wincred works and that starting from scratch works.

File: go/libkb/secret_store_windows.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+				continue
+			}
+			fileStore.ClearSecret(NormalizedUsername(fileUser))
+		}
+	}
+	return nil
+}
+
+func NewSecretStoreAll(g *GlobalContext) SecretStoreAll {
+	if g.Env.ForceSecretStoreFile() {
```
Comment (patrickxb) [inline go/libkb/secret_store_windows.go]: What about only in admin builds as well for now?

**Comment by patrickxb** [comment]
Is this available on all versions of windows?

**Comment by zanderz** [comment]
These functions are Windows XP and later: https://msdn.microsoft.com/en-us/library/windows/desktop/aa374804%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396

@patrickxb What is an admin build exactly? `devel` tag? lack of `production` tag? I will make a production build off this branch right now and sanity check it.

**Comment by patrickxb** [comment]
Admin builds are the internal builds that we get daily.  I assume those happen for windows too?

You can check `KEYBASE_FEATURES=admin` via G.Env.GetFeatureFlags().

**Comment by zanderz** [comment]
The admin/test channel builds are delivered per updater ID, which is set on the server side, and are not compiled or run any differently otherwise AFAIK, at least on Windows. Or if they're supposed to be, they're not now.

I see where the environment or configuration is checked for a string in the list of feature flags called "admin" in libkb/features.go, but no code calls FeatureFlags.Admin() - I will add a check for that though.

The problem with building just this branch in the test channel is that automatic daily builds off Master also go in there, so people will get updated back off of this build and not have their .ss files any more.

**Comment by patrickxb** [comment]
did something happen in git for this branch?  55 files changed...

**Comment by zanderz** [comment]
I merged Master in, probably that was it. Stripped it back out for now.

Build is http://s3.amazonaws.com/prerelease.keybase.io/windows/Keybase_1.0.44-20180226115625%2B696302344.386.exe if anyone wants to try - works fine for me.

**Comment by zanderz** [comment]
can I merge this?

**Pull request merged**